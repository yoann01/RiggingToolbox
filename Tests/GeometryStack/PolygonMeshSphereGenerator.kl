
require RiggingToolbox;

operator entry(){
  PolygonMeshSphereGenerator sphereGenerator();

  EvalContext context();
  GeometrySet geomSet();
  sphereGenerator.evaluate(context, geomSet);

  report("NumGeometries:" + geomSet.size);
  for(Integer i=0; i<geomSet.size; i++){
    report(geomSet.get(i).getDesc(false, false));
  }
}

