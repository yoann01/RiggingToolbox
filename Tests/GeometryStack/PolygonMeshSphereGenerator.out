NumGeometries:1
Mesh: pointCount: 57 polygonCount: 66 nbAttributeVectors: 105
  Points (adjacent polygons as 'polygon.neighborPolygonIndex',
          borders as '|', closed wing starts as '<<'):
    0: 11 polygons:  <<10.0, 9.0, 8.0, 7.0, 6.0, 5.0, 4.0, 3.0, 2.0, 1.0, 0.0
    1: 4 polygons:  <<21.1, 10.1, 0.2, 11.0
    2: 4 polygons:  <<12.0, 11.1, 0.1, 1.2
    3: 4 polygons:  <<13.0, 12.1, 1.1, 2.2
    4: 4 polygons:  <<14.0, 13.1, 2.1, 3.2
    5: 4 polygons:  <<15.0, 14.1, 3.1, 4.2
    6: 4 polygons:  <<16.0, 15.1, 4.1, 5.2
    7: 4 polygons:  <<17.0, 16.1, 5.1, 6.2
    8: 4 polygons:  <<18.0, 17.1, 6.1, 7.2
    9: 4 polygons:  <<19.0, 18.1, 7.1, 8.2
    10: 4 polygons:  <<20.0, 19.1, 8.1, 9.2
    11: 4 polygons:  <<21.0, 20.1, 9.1, 10.2
    12: 4 polygons:  <<32.1, 21.2, 11.3, 22.0
    13: 4 polygons:  <<23.0, 22.1, 11.2, 12.3
    14: 4 polygons:  <<24.0, 23.1, 12.2, 13.3
    15: 4 polygons:  <<25.0, 24.1, 13.2, 14.3
    16: 4 polygons:  <<26.0, 25.1, 14.2, 15.3
    17: 4 polygons:  <<27.0, 26.1, 15.2, 16.3
    18: 4 polygons:  <<28.0, 27.1, 16.2, 17.3
    19: 4 polygons:  <<29.0, 28.1, 17.2, 18.3
    20: 4 polygons:  <<30.0, 29.1, 18.2, 19.3
    21: 4 polygons:  <<31.0, 30.1, 19.2, 20.3
    22: 4 polygons:  <<32.0, 31.1, 20.2, 21.3
    23: 4 polygons:  <<43.1, 32.2, 22.3, 33.0
    24: 4 polygons:  <<34.0, 33.1, 22.2, 23.3
    25: 4 polygons:  <<35.0, 34.1, 23.2, 24.3
    26: 4 polygons:  <<36.0, 35.1, 24.2, 25.3
    27: 4 polygons:  <<37.0, 36.1, 25.2, 26.3
    28: 4 polygons:  <<38.0, 37.1, 26.2, 27.3
    29: 4 polygons:  <<39.0, 38.1, 27.2, 28.3
    30: 4 polygons:  <<40.0, 39.1, 28.2, 29.3
    31: 4 polygons:  <<41.0, 40.1, 29.2, 30.3
    32: 4 polygons:  <<42.0, 41.1, 30.2, 31.3
    33: 4 polygons:  <<43.0, 42.1, 31.2, 32.3
    34: 4 polygons:  <<54.1, 43.2, 33.3, 44.0
    35: 4 polygons:  <<45.0, 44.1, 33.2, 34.3
    36: 4 polygons:  <<46.0, 45.1, 34.2, 35.3
    37: 4 polygons:  <<47.0, 46.1, 35.2, 36.3
    38: 4 polygons:  <<48.0, 47.1, 36.2, 37.3
    39: 4 polygons:  <<49.0, 48.1, 37.2, 38.3
    40: 4 polygons:  <<50.0, 49.1, 38.2, 39.3
    41: 4 polygons:  <<51.0, 50.1, 39.2, 40.3
    42: 4 polygons:  <<52.0, 51.1, 40.2, 41.3
    43: 4 polygons:  <<53.0, 52.1, 41.2, 42.3
    44: 4 polygons:  <<54.0, 53.1, 42.2, 43.3
    45: 4 polygons:  <<65.2, 54.2, 44.3, 55.1
    46: 4 polygons:  <<56.1, 55.2, 44.2, 45.3
    47: 4 polygons:  <<57.1, 56.2, 45.2, 46.3
    48: 4 polygons:  <<58.1, 57.2, 46.2, 47.3
    49: 4 polygons:  <<59.1, 58.2, 47.2, 48.3
    50: 4 polygons:  <<60.1, 59.2, 48.2, 49.3
    51: 4 polygons:  <<61.1, 60.2, 49.2, 50.3
    52: 4 polygons:  <<62.1, 61.2, 50.2, 51.3
    53: 4 polygons:  <<63.1, 62.2, 51.2, 52.3
    54: 4 polygons:  <<64.1, 63.2, 52.2, 53.3
    55: 4 polygons:  <<65.1, 64.2, 53.2, 54.3
    56: 11 polygons:  <<65.0, 55.0, 56.0, 57.0, 58.0, 59.0, 60.0, 61.0, 62.0, 63.0, 64.0
  Polygons (connected points as 'point.polygonPointIndex', borders as '|'):
    0: 3 points: 0.10, 2.2, 1.2
    1: 3 points: 0.9, 3.2, 2.3
    2: 3 points: 0.8, 4.2, 3.3
    3: 3 points: 0.7, 5.2, 4.3
    4: 3 points: 0.6, 6.2, 5.3
    5: 3 points: 0.5, 7.2, 6.3
    6: 3 points: 0.4, 8.2, 7.3
    7: 3 points: 0.3, 9.2, 8.3
    8: 3 points: 0.2, 10.2, 9.3
    9: 3 points: 0.1, 11.2, 10.3
    10: 3 points: 0.0, 1.1, 11.3
    11: 4 points: 1.3, 2.1, 13.2, 12.2
    12: 4 points: 2.0, 3.1, 14.2, 13.3
    13: 4 points: 3.0, 4.1, 15.2, 14.3
    14: 4 points: 4.0, 5.1, 16.2, 15.3
    15: 4 points: 5.0, 6.1, 17.2, 16.3
    16: 4 points: 6.0, 7.1, 18.2, 17.3
    17: 4 points: 7.0, 8.1, 19.2, 18.3
    18: 4 points: 8.0, 9.1, 20.2, 19.3
    19: 4 points: 9.0, 10.1, 21.2, 20.3
    20: 4 points: 10.0, 11.1, 22.2, 21.3
    21: 4 points: 11.0, 1.0, 12.1, 22.3
    22: 4 points: 12.3, 13.1, 24.2, 23.2
    23: 4 points: 13.0, 14.1, 25.2, 24.3
    24: 4 points: 14.0, 15.1, 26.2, 25.3
    25: 4 points: 15.0, 16.1, 27.2, 26.3
    26: 4 points: 16.0, 17.1, 28.2, 27.3
    27: 4 points: 17.0, 18.1, 29.2, 28.3
    28: 4 points: 18.0, 19.1, 30.2, 29.3
    29: 4 points: 19.0, 20.1, 31.2, 30.3
    30: 4 points: 20.0, 21.1, 32.2, 31.3
    31: 4 points: 21.0, 22.1, 33.2, 32.3
    32: 4 points: 22.0, 12.0, 23.1, 33.3
    33: 4 points: 23.3, 24.1, 35.2, 34.2
    34: 4 points: 24.0, 25.1, 36.2, 35.3
    35: 4 points: 25.0, 26.1, 37.2, 36.3
    36: 4 points: 26.0, 27.1, 38.2, 37.3
    37: 4 points: 27.0, 28.1, 39.2, 38.3
    38: 4 points: 28.0, 29.1, 40.2, 39.3
    39: 4 points: 29.0, 30.1, 41.2, 40.3
    40: 4 points: 30.0, 31.1, 42.2, 41.3
    41: 4 points: 31.0, 32.1, 43.2, 42.3
    42: 4 points: 32.0, 33.1, 44.2, 43.3
    43: 4 points: 33.0, 23.0, 34.1, 44.3
    44: 4 points: 34.3, 35.1, 46.2, 45.2
    45: 4 points: 35.0, 36.1, 47.2, 46.3
    46: 4 points: 36.0, 37.1, 48.2, 47.3
    47: 4 points: 37.0, 38.1, 49.2, 48.3
    48: 4 points: 38.0, 39.1, 50.2, 49.3
    49: 4 points: 39.0, 40.1, 51.2, 50.3
    50: 4 points: 40.0, 41.1, 52.2, 51.3
    51: 4 points: 41.0, 42.1, 53.2, 52.3
    52: 4 points: 42.0, 43.1, 54.2, 53.3
    53: 4 points: 43.0, 44.1, 55.2, 54.3
    54: 4 points: 44.0, 34.0, 45.1, 55.3
    55: 3 points: 56.1, 45.3, 46.1
    56: 3 points: 56.2, 46.0, 47.1
    57: 3 points: 56.3, 47.0, 48.1
    58: 3 points: 56.4, 48.0, 49.1
    59: 3 points: 56.5, 49.0, 50.1
    60: 3 points: 56.6, 50.0, 51.1
    61: 3 points: 56.7, 51.0, 52.1
    62: 3 points: 56.8, 52.0, 53.1
    63: 3 points: 56.9, 53.0, 54.1
    64: 3 points: 56.10, 54.0, 55.1
    65: 3 points: 56.0, 55.0, 45.0

